pravega/pravega
# Issue 326: project restructure

**Change log description**
Relates to issue #326 and #1263.
This restructures the projects to avoid conflicting project names, allowing the group to be set to produce correct POM dependencies.

Rename summary:
- `:controller:server` -> `:controller`
- `:service:*` -> `:segmentstore:*`
- `:shared:controller` -> `:shared:controller-api`
- `:test:common` -> `:test:testcommon`

Note that renaming the projects also cause the output artifact names to change accordingly:
- `pravega-controller-server` -> `pravega-controller`
- `pravega-service-*` -> `pravega-segmentstore-*`
- `pravega-shared-controller` -> `pravega-shared-controller-api`
- `pravega-test-common` -> `pravega-test-testcommon`

**Purpose of the change**
To fix the POM dependencies and to rename some of the projects to a more clear name.

**What the code does**
No code changes, only structural changes.

**How to verify it**
`./gradlew clean install`, check the POM dependencies and see that all dependencies have a group ID of `io.pravega`
